### PR TITLE
[FEATURE] builder: Improve support for ES6+ syntax 

### PR DESCRIPTION
**Pull request by matz3**
JIRA: CPOUI5FOUNDATION-374

### CODE REVIEW

File: test/lib/lbt/calls/SapUiDefine.js [hunk 1]
```diff
@@ -1,13 +1,32 @@
...
+const loggerInstance = logger.getLogger();
+const sinonGlobal = require("sinon");
+const mock = require("mock-require");
 
 function parse(code) {
 	const ast = parseJS(code);
 	return ast.body[0].expression;
 }
 
+test.before((t) => {
```
Comment (matz3) [inline test/lib/lbt/calls/SapUiDefine.js:14]: shouldn't this be beforeEach? (same for the other tests)

File: test/lib/lbt/calls/SapUiDefine.js [hunk 2]
```diff
@@ -1,13 +1,32 @@
...
+	const sinon = t.context.sinon = sinonGlobal.createSandbox();
+	t.context.warningLogSpy = sinon.spy(loggerInstance, "warn");
+	sinon.stub(logger, "getLogger").returns(loggerInstance);
+	t.context.SapUiDefineCallWithStubbedLogger = mock.reRequire("../../../../lib/lbt/calls/SapUiDefine");
+});
+
+test.after((t) => {
```
Comment (matz3) [inline test/lib/lbt/calls/SapUiDefine.js:21]: and afterEach here? (same for the other tests)

Comment (flovogt) [reply test/lib/lbt/calls/SapUiDefine.js:21]: Done

File: test/lib/lbt/calls/SapUiDefine.js (hunk 1 above)
Comment (flovogt) [reply test/lib/lbt/calls/SapUiDefine.js:14]: Done

File: test/lib/lbt/calls/SapUiDefine.js (hunk 2 above)
Comment (flovogt) [reply test/lib/lbt/calls/SapUiDefine.js:21]: Done

File: lib/lbt/analyzer/FioriElementsAnalyzer.js [hunk 3]
```diff
@@ -181,6 +181,10 @@ class FioriElementsAnalyzer {
 						}
 					});
 				}
+				if (defineCall.factory.async) {
```
[the remark this answers is not in the document]
Comment (flovogt) [reply lib/lbt/analyzer/FioriElementsAnalyzer.js:184]: Thats true. Done

File: lib/lbt/analyzer/SmartTemplateAnalyzer.js [hunk 4]
```diff
@@ -134,24 +134,39 @@ class TemplateComponentAnalyzer {
...
+					defineCall.factory.body.body.forEach( (stmt) => {
+						if ( stmt.type === Syntax.ReturnStatement &&
+							this._isTemplateClassDefinition(TA, stmt.argument)
+						) {
+							templateName =
+								this._analyzeTemplateClassDefinition(stmt.argument.arguments[2]) || templateName;
+						}
+					});
+				}
+				if (defineCall.factory.async) {
```
Comment (codeworrior) [inline lib/lbt/analyzer/SmartTemplateAnalyzer.js:153]: This message is IMO misleading. The ui5loader can handle async callbacks. The returned promise is just used as export value and delivered to anyone importing the module. 

Here, the smart templates (FE v2) cannot handle such a promise. That's the reason why we should log a message.

Comment (matz3) [reply lib/lbt/analyzer/SmartTemplateAnalyzer.js]: Let's handle this in a follow-up PR, so that we can merge the current changes.

**Review by matz3 — COMMENTED** [review]
LGTM 👍🏻

**Review by RandomByte — APPROVED** [review]
Had a rudimentary look, and LGTM

**Pull request merged**